test/Unit/Porter/Net/Caching/CachingConnectorTest.php

```diff
@@ -0,0 +1,29 @@
+<?php
+namespace ScriptFUSIONTest\Unit\Porter\Net\Caching;
+
+use Mockery\Adapter\Phpunit\MockeryPHPUnitIntegration;
+use ScriptFUSION\Porter\Cache\MemoryCache;
+
+final class CachingConnectorTest extends \PHPUnit_Framework_TestCase
+{
+    use MockeryPHPUnitIntegration;
+
+    public function test()
+    {
+        $memoryCache = new MemoryCache();
+
+        $connector = new CachingTestConnector();
+        $connector->setCache($memoryCache);
+        $source = '@foo@';
+
+        $structure = [$source, []];
+        self::assertSame('bar', $connector->fetch($source, null));
+
+        $hash = str_replace(str_split('{}()/\@:'), '.', json_encode($structure));
```

Comment:
Seeing business logic duplicated in a test is usually an indication of a problem. In this case, the problem seems to be we're testing the exact output of the hashing function instead of just what we really care about: **the banned characters are excluded**. This creates a very brittle test because any change to the code causes the test to fail. 

Let's just test that the hash string doesn't contain the banned characters, so if the implementation changes, the test should continue to pass.